File: kafka-helper-domain/src/main/java/com/hg/kafka/helper/domain/repository/HostInfoRepository.java
package com.hg.kafka.helper.domain.repository;

import com.hg.kafka.helper.domain.entity.HostInfo;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Date;
import java.util.List;
import java.util.Optional;

@Repository
public interface HostInfoRepository extends MongoRepository<HostInfo, String> {
    List<HostInfo> findAllByHashkey(String hashkey);
    Optional<HostInfo> findOneByHostName(String hostName);
    List<HostInfo> findAllByUpdatedAtLessThan(Date UpdatedAt);
}

File: kafka-helper-domain/src/main/kotlin/com/hg/kafka/helper/domain/repository/HostInfoRepository.kt
package com.hg.kafka.helper.domain.repository

import com.hg.kafka.helper.domain.entity.HostInfo
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface HostInfoRepository : MongoRepository<HostInfo, String> {
    fun findAllByHashkey(hashkey: String?): List<HostInfo>
    fun findOneByHostName(hostName: String?): Optional<HostInfo>
    fun findAllByUpdatedAtLessThan(UpdatedAt: Date): List<HostInfo>
}
